documentation/guide-jpa-query.asciidoc

```diff
@@ -25,42 +25,84 @@ SELECT line FROM OrderLineEntity line WHERE line.order.id = :orderId
     if ((name != null) && (!name.isEmpty())) {
-      // query.where(Alias.$(alias.getName()).eq(name));
-      QueryUtil.get().whereString(query, Alias.$(alias.getName()), name, criteria.getNameOption());
+      query.where(dish.name.eq(name));
     }
     return query.fetch();
   }
 ----
 
+In this example we use the so called Q-types (`QDishEntity`). These are classes generated at build time by the QueryDSL annotation processor from entity classes. The Q-type classes can be used as static types representative of the original entity class.
+
+For spring, devon4j provides another approach that you can use for your Spring applications to implement QueryDSL logic without having to use these metaclasses. An example can be found link:spring/guide-querydsl-spring.asciidoc[here].
+
 == Using Wildcards
 For flexible queries it is often required to allow wildcards (especially in xref:dynamic_queries[dynamic queries]). While users intuitively expect glob syntax the SQL and JPQL standards work different. Therefore a mapping is required. devonfw provides this on a lower level by https://github.com/devonfw/devon4j/blob/develop/modules/basic/src/main/java/com/devonfw/module/basic/common/api/query/LikePatternSyntax.java[LikePatternSyntax] and on a high level by https://github.com/devonfw/devon4j/blob/develop/modules/jpa-basic/src/main/java/com/devonfw/module/jpa/dataaccess/api/QueryUtil.java#L54[QueryUtil] (see https://github.com/devonfw/devon4j/blob/develop/modules/jpa-basic/src/main/java/com/devonfw/module/jpa/dataaccess/api/QueryHelper.java#L199[QueryHelper.newStringClause(...)]).
 
 == Pagination
-devonfw provides pagination support. If you are using link:guide-repository.asciidoc[spring-data repositories] you will get that directly from spring for static queries. Otherwise for dynamic or generally handwritten queries we provide this via https://github.com/devonfw/devon4j/blob/develop/modules/jpa-basic/src/main/java/com/devonfw/module/jpa/dataaccess/api/QueryUtil.java#L102[QueryUtil.findPaginated(...)]:
+When dealing with large amounts of data, an efficient method of retrieving the data is required. Fetching the entire data set each time would be too time consuming. Instead, __Paging__ is used to process only small subsets of the entire data set.
+
+If you are using link:guide-repository.asciidoc[Spring Data repositories] you will get pagination support out of the box by providing the interfaces https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/Pageable.html[Page] and  https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/Pageable.html[Pageable]:
+
+.**repository**
+[source,java]
+----
+Page<DishEntity> findAll(Pageable pageable);
+----
+
+Then you can create a Pageable object and pass it to the method call as follows:
+
+[source,java]
+----
+int page = criteria.getPageNumber();
+int size = criteria.getPageSize();
+Pageable pageable = PageRequest.of(page, size);
```

Comment:
What we typically have as criteria already is or contains a pagable (e.g. using `AbstractSearchCriteriaTo`).
Therefore not the best example, but nice to have that we can also improve later so nothing that prevents merge of PR...